android/gltfio-android/src/main/cpp/MaterialKey.cpp

```diff
@@ -0,0 +1,103 @@
+/*
+ * Copyright (C) 2021 The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <jni.h>
+
+#include <gltfio/MaterialProvider.h>
+
+#include "MaterialKey.h"
+
+using namespace gltfio;
+
+void nativeFromJava(JNIEnv* env, MaterialKey& dst, jobject src) {
```

Comment:
There are many fields here, so it would be like over 30 arguments, are we sure that's an okay thing to do?  Agreed that I should at least cache all the field ids.

Reply:
I think it's okay, and it'll prevent mistakes (like typos) by failing at compile time.  From a performance standpoint (and maybe even code size), it would probably be better. 
But I realize that this is called from native -- so if you're going to do a native->java transition for that, it's probably not worth it. However, if you already have a switch to java, you should do as much as possible there.

Reply:
@pixelflinger okay I added a commit that adds a helper class for caching these field ids.  Since the material key is an in-out argument, I think this is a better solution than a long argument list.